packages/xstate-solid/src/createImmutable.ts

```diff
@@ -53,7 +53,7 @@ const updateStore = <Path extends unknown[]>(
       const largestSize = Math.max(next.length, prev.length);
 
       // Diff array
-      for (let start = 0, end = smallestSize - 1; start < end; start++, end--) {
+      for (let start = 0, end = largestSize - 1; start < end; start++, end--) {
```

Comment:
could you explain what happens here and what was the meaning of the old code + what is the meaning of the new one?

Reply:
This loops through and diffs each index of the old and new array. The old code only diffed between index 0 and the length of the smallest array, because in the next if statement, we handle any new or now undefined indexes in the new array.

The new code diffs between index 0 and the length of the largest array. This will always diff and update the entire array in the loop; then the if statement will ensure the length and new indexes are correct. The new code may hit some indexes twice in diffing, but it won't affect performance in any meaningful way, and we shouldn't see any duplicate array indexes anymore.